content/PreparedStatement/PreparedStatement.q.md

```diff
@@ -0,0 +1,1708 @@
+
+- It assigns the named placeholders `:first_name`, `:last_name`, `:birth_date`, and `:gender` to be inserted into the `students` table.
+
+- It sets the positional placeholders `:first_name`, `:last_name`, `:birth_date`, and `:gender` to replace the values in the SQL statement.
+
+- It replaces the values in the SQL statement with named placeholders `:first_name`, `:last_name`, `:birth_date`, and `:gender` to have randomized values.
+
+- It substitutes the values in the SQL statement with named placeholders `:first_name`, `:last_name`, `:birth_date`, and `:gender` to prevent from SQL injections.
+
+
+/// type=MS, answer=[2,3,4,5]
+
+Which statements correctly describe `$pstmt = $conn->prepare($sql)` line 12 of `student.php`?
+
+- It executes the SQL statement.
+
+- It prepares the argument `$sql` for execution.
+
+- It returns the result set as a `PDOStatement` object.
+
+- It calls the `prepare()` method of the `$conn` object.
+
+- It assigns the `PDOStatement` object to the variable `$pstmt`.
+
+
+:::
+
+
+:::
+
+/// type=REPL, readonly=true, filename=[connection.php,student.php]
+
+```php
+// connection.php
+<?php
+    $host = 'localhost';
+    $db = 'LibraryDB';
+    $port = '5432';
+    $username = 'postgres';
+    $password = 'Admin01';
+
+    try {
+        $conn = new PDO("pgsql:host=$host;port=$port;dbname=$db;user=$username;password=$password");
+        if ($conn) {
+            echo "Successfully connected to the database. <br />";
+        }
+    } catch (Exception $e) {
+        echo "Unable to establish a connection."; 
+    }
+?>
+```
+
+```php
+// student.php
+<?php
+    require_once("connection.php");
+
+    $pstmt = $conn->prepare("SELECT * FROM students WHERE gender = ?");
+    $pstmt->execute(array('Female'));
+    $data = $pstmt->fetchAll();
+    
+    foreach ($data as $row) {
+        echo $row['student_id'].$row['first_name'].$row['last_name'].$row['birth_date'].$row['gender']. ' ';
+    }
+?>
+```
+/// type=SS, answer=[5]
+
+Execute the program. What is its output?
+
+- It produces an error.
+
+- No output is displayed.
+
+- It prints `Unable to establish a connection.`.
+
+- It prints `Successfully connected to the database.` and `2SamanthaDanes1999-10-12Female 4VikkiFernando2005-05-17Female `.
+
+- It prints `Successfully connected to the database.` and `2SamanthaDanes1999-10-12Female 4VikkiFernando2005-05-17Female 6AlisaElls1999-06-30Female `.
+
+
+/// type=MS, answer=[2,3,4,5]
+
+Which statements correctly describe the code on line 4 of `student.php`?
+
+- It queries the SQL statement.
+
+- It returns the `PDOStatement` object.
+
+- It calls the `prepare()` method of the `$conn` object.
+
+- It assigns the `PDOStatement` object to the variable `$pstmt`.
+
+- It prepares the argument `SELECT * FROM students WHERE gender = ?` in the `prepare()` method.
+
+
+/// type=MS, answer=[1,5]
+
+Which statements correctly describe `WHERE gender = ?` on line 4 of `student.php`?
```

Comment:
Please check what `WHERE gender = ?` is all about for you to easily describe this correctly. Thanks.